src/common/components/dateTimeTextField/DateTimeTextField.tsx

```diff
@@ -12,52 +12,69 @@ const useStyles = makeStyles({
   textFieldInput: { width: '100%' },
 });
 
+/**
+ * Removes the special prefix `@@react-admin@@` and the needless quotation marks
+ * that might be added by the useInput-hook.
+ * @param message the translation key (with ra-core's special prefix to avoid double translation)
+ * @returns the fixed translation key
+ * @example
+ *    // a code related to the validation in the useInput.ts...
+ *    validate: async (value, values) => {
+ *        if (!sanitizedValidate) return true;
+ *        const error = await sanitizedValidate(value, values, props);
+ *        if (!error) return true;
+ *        // react-hook-form expects errors to be plain strings but our validators can return objects
+ *        // that have message and args.
+ *        // To avoid double translation for users that validate with a schema instead of our validators
+ *        // we use a special format for our validators errors.
+ *        // The ValidationError component will check for this format and extract the message and args
+ *        // to translate.
+ *        return `@@react-admin@@${JSON.stringify(error)}`;
+ */
+const handleUseInputDoubleValidationMessage = (message: string) =>
+  message.replace('@@react-admin@@', '').replaceAll('"', '');
```

Comment:
If the quotation marks should only be removed from the start and the end then how about something like this:
```typescript
.replace(/^"(.*)"$/, '$1'); // Remove the possible quotation marks around the string
```
or not as accurately (i.e. doesn't require the quotation marks to be in pairs):
```typescript
.replace(/^"/, '') // Remove possible leading quotation mark
.replace(/"$/, '') // Remove possible trailing quotation mark
```

Reply:
well, the json key cannot have quotation marks in the middle